modules/RTC/RTCUtils.js

```diff
@@ -555,6 +555,49 @@ function enumerateDevicesThroughMediaStreamTrack(callback) {
 }
 
 /**
+ * Returns a promise to check whether or not the permission prompt for user
+ * media might display. The promise returns a boolean but the boolean value is
+ * not a definitive indicator, except in the case of false as then it is certain
+ * no prompt will display.
+ *
+ * @param {Function} enumerateDevices - The native enumerate devices function.
+ * @returns {Promise<boolean>}
+ */
+function mightShowPermissionPrompt(enumerateDevices) {
+    // The focus check is necessary for browsers (Firefox) that do not complete
+    // getUserMedia when the browser tab is not in focus. In those cases, it is
+    // not known with certainty if the permission prompt will display or not.
+    if (!document.hasFocus()) {
```

Comment:
In this case what will be the result of enumerateDevices call. It will not finish too?

It's not clear for me what is the connection between the prompt detection via `enumerateDevices` and the focused tab. In the comment you are mentioning GUM and I don't see how GUM is related to this function.